File: UVa/400/412.py
from fractions import *
from math import *

while True:
    n = int(input())
    if n == 0:
        exit()

    v = []
    num_gcd = 0
    for i in range(n):
        v.append(int(input()))
        for j in range(i):
            if gcd(v[i], v[j]) == 1:
                num_gcd += 1

    if num_gcd == 0:
        print("No estimate for this data set.");
    else:
        print("%.6f" % sqrt(3 * n * (n - 1) / num_gcd));

File: UVa/400/412.cpp
#include <bits/stdc++.h>
#include "header.h"

int main() {
	int n;
	while (cin >> n && n != 0) {
		vi nums(n);
		int count = 0;
		read(nums);
		drep(i, j, n, i) {
			if (gcd(nums[i], nums[j]) == 1)
				count++;
		}
		if (count == 0) {
			print("No estimate for this data set.");
			continue;
		}
		double ans2 = 3. / count * n * (n - 1);
		print("%.6f", sqrt(ans2));
	}
}
